public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx

```diff
@@ -159,7 +159,8 @@ export const QueryAndExpressionsStep = ({ editingExistingRule, onDataChange }: P
     if (!editingExistingRule && isOptimizeReducerEnabled) {
       dispatch(optimizeReduceExpression({ updatedQueries: dataQueries, expressionQueries }));
     }
-  }, [dataQueries, expressionQueries, editingExistingRule, isOptimizeReducerEnabled]);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
we need to keep this useEffect only for the first render. Otherwise, the reducer is not optimized.

Reply:
ohhh yesss!!!!